cmd/podman/images/scp.go

```diff
@@ -0,0 +1,350 @@
+func createConnections(uri []*urlP.URL, iden []string) ([]*ssh.Client, []string, error) {
+	connList := []*ssh.Client{}
+	remoteFiles := []string{}
+	for i := range uri {
+		cfg, err := connection.ValidateAndConfigure(uri[i], iden[i])
+		if err != nil {
+			return nil, nil, err
+		}
+		dialAdd, err := ssh.Dial("tcp", uri[0].Host, cfg) // dial the client
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "failed to connect")
+		}
+		connList = append(connList, dialAdd)
+		file, err := makeRemoteFile(dialAdd)
+		if err != nil {
+			return nil, nil, err
+		}
+		remoteFiles = append(remoteFiles, file)
+	}
+	return connList, remoteFiles, nil
+}
+
+// parseLocation returns the valid connection data based off of the information provided by the user
+// iden is a string array of ssh identity keys, url is a string array of paths to ssh clients. The first entry
+// in each array is the save location and the second entry is the load location if applicable. An empty string will be
+// inserted if identity is blank (not required in some cases).
+// serv is a map of service destinations with the connection name as the index intended to be used as EngineConfig.ServiceDestinations
+// this function modifies the global scpOpt entities: FromRemote, ToRemote, Connections, and DestImageName
+func parseArgs(args []string, cfg *config.Config) ([]string, []string, map[string]config.Destination, error) {
+	iden := []string{}
+	url := []string{}
+	serv := map[string]config.Destination{}
+	cliConnections := []string{}
+	switch len(args) {
+	case 1:
+		if strings.Contains(args[0], "::") {
+			scpOpts.FromRemote = true
+			cliConnections = append(cliConnections, args[0])
+		} else {
+			scpOpts.DestImageName = args[0]
+			named, err := reference.ParseNormalizedNamed(scpOpts.DestImageName)
+			if err != nil {
+				return nil, nil, nil, err
+			}
+			scpOpts.DestImageName = named.String()
+		}
+	case 2:
+		if strings.Contains(args[0], "::") {
+			if !(strings.Contains(args[1], "::")) && len(strings.Split(args[0], "::")) != 2 { // if an image is specified, this mean we are loading from our client
+				cliConnections = append(cliConnections, args[0])
+				scpOpts.ToRemote = true
+			} else if strings.Contains(args[1], "::") { // both remote clients
+				scpOpts.FromRemote = true
+				scpOpts.ToRemote = true
+				if len(strings.Split(args[0], "::")) == 1 || (len(strings.Split(args[0], "::")) == 2 && len(strings.Split(args[1], "::")) == 2) { // is save->load w/ one image name or save-load w/ two image names
+					cliConnections = append(cliConnections, args[0])
+					cliConnections = append(cliConnections, args[1])
+				} else { // else its a load save (order of args) or invalid
+					cliConnections = append(cliConnections, args[1])
+					cliConnections = append(cliConnections, args[0])
+				}
+			} else {
+				return nil, nil, nil, errors.Wrapf(define.ErrInvalidArg, "cannot specify host::image image")
```

Comment:
If this is supposed to be prohibited, shouldn’t that happen in the `ToRemote && !FromRemote` case above?

Reply:
no, the `ToRemote && !FromRemote` is the case of length 1. In this case: you have two arguments so that will never happen, this is basically a fail safe. Before it was working if you did `image scp fed::alpine FOOBAR` but just ignoring `FOOBAR` and loading alpine to fed. Now it gives an error as it should.